### ISSUE TITLE
[Crane] java.util.NoSuchElementException: Key 781.0 is missing in the map.

### ISSUE THREAD

**Issue by karan4c6** [android/compose-samples#408]
2021-02-25 11:13:15.313 5798-5798/androidx.compose.samples.crane E/AndroidRuntime: FATAL EXCEPTION: main
    Process: androidx.compose.samples.crane, PID: 5798
    java.util.NoSuchElementException: Key 781.0 is missing in the map.
        at kotlin.collections.MapsKt__MapWithDefaultKt.getOrImplicitDefaultNullable(MapWithDefault.kt:24)
        at kotlin.collections.MapsKt__MapsKt.getValue(Maps.kt:344)
        at androidx.compose.material.SwipeableKt$swipeable$3$3$1.invoke(Swipeable.kt:578)
        at androidx.compose.material.SwipeableKt$swipeable$3$3$1.invoke(Swipeable.kt:577)
        at androidx.compose.material.SwipeableKt.computeTarget(Swipeable.kt:765)
        at androidx.compose.material.SwipeableKt.access$computeTarget(Swipeable.kt:1)
        at androidx.compose.material.SwipeableState.getTargetValue(Swipeable.kt:223)
        at androidx.compose.material.BackdropScaffoldKt$BackdropScaffold$1$1$2.invoke(BackdropScaffold.kt:342)
        at androidx.compose.material.BackdropScaffoldKt$BackdropScaffold$1$1$2.invoke(BackdropScaffold.kt:333)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.material.SurfaceKt$Surface$1.invoke(Surface.kt:116)
        at androidx.compose.material.SurfaceKt$Surface$1.invoke(Surface.kt:105)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:193)
        at androidx.compose.material.SurfaceKt.Surface-F-jzlyU(Surface.kt:102)
        at androidx.compose.material.BackdropScaffoldKt$BackdropScaffold$1$1.invoke-jYbf7pk(BackdropScaffold.kt:325)
        at androidx.compose.material.BackdropScaffoldKt$BackdropScaffold$1$1.invoke(BackdropScaffold.kt:298)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:149)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.material.BackdropScaffoldKt$BackdropStack$1$1$placeables$1.invoke(BackdropScaffold.kt:447)
        at androidx.compose.material.BackdropScaffoldKt$BackdropStack$1$1$placeables$1.invoke(BackdropScaffold.kt:446)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.ui.layout.SubcomposeLayoutState$subcompose$2$1.invoke(SubcomposeLayout.kt:167)
        at androidx.compose.ui.layout.SubcomposeLayoutState$subcompose$2$1.invoke(SubcomposeLayout.kt:167)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.ComposerKt.invokeComposable(Composer.kt:3418)
        at androidx.compose.runtime.ComposerImpl.composeContent$runtime_release(Composer.kt:2600)
        at androidx.compose.runtime.CompositionImpl.composeContent(Composition.kt:348)
        at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(Recomposer.kt:693)
        at androidx.compose.runtime.ComposerImpl$CompositionContextImpl.composeInitial$runtime_release(Composer.kt:3024)
        at androidx.compose.runtime.ComposerImpl$CompositionContextImpl.composeInitial$runtime_release(Composer.kt:3024)
        at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:304)
        at androidx.compose.ui.layout.SubcomposeLayoutState.subcomposeInto(SubcomposeLayout.kt:184)
        at androidx.compose.ui.layout.SubcomposeLayoutState.access$subcomposeInto(SubcomposeLayout.kt:100)
        at androidx.compose.ui.layout.SubcomposeLayoutState$subcompose$2.invoke(SubcomposeLayout.kt:160)
        at androidx.compose.ui.layout.SubcomposeLayoutState$subcompose$2.invoke(SubcomposeLayout.kt:158)
        at androidx.compose.runtime.snapshots.SnapshotStateObserver.withNoObservations(SnapshotStateObserver.kt:137)
2021-02-25 11:13:15.314 5798-5798/androidx.compose.samples.crane E/AndroidRuntime:     at androidx.compose.ui.node.OwnerSnapshotObserver.withNoSnapshotReadObservation$ui_release(OwnerSnapshotObserver.kt:49)
        at androidx.compose.ui.node.LayoutNode.withNoSnapshotReadObservation$ui_release(LayoutNode.kt:1085)
        at androidx.compose.ui.layout.SubcomposeLayoutState.subcompose(SubcomposeLayout.kt:158)
        at androidx.compose.ui.layout.SubcomposeLayoutState.subcompose(SubcomposeLayout.kt:152)
        at androidx.compose.material.BackdropScaffoldKt$BackdropStack$1$1.invoke-0kLqBqw(BackdropScaffold.kt:446)
        at androidx.compose.material.BackdropScaffoldKt$BackdropStack$1$1.invoke(BackdropScaffold.kt:438)
        at androidx.compose.ui.layout.SubcomposeLayoutState$createMeasurePolicy$1.measure-3p2s80s(SubcomposeLayout.kt:212)
        at androidx.compose.ui.node.InnerPlaceable.performMeasure-BRTryo0(InnerPlaceable.kt:45)
        at androidx.compose.ui.node.LayoutNodeWrapper.measure-BRTryo0(LayoutNodeWrapper.kt:156)
        at androidx.compose.foundation.layout.FillModifier.measure-3p2s80s(Size.kt:647)
        at androidx.compose.ui.node.ModifiedLayoutNode.performMeasure-BRTryo0(ModifiedLayoutNode.kt:36)
        at androidx.compose.ui.node.LayoutNodeWrapper.measure-BRTryo0(LayoutNodeWrapper.kt:156)
        at androidx.compose.ui.node.OuterMeasurablePlaceable$remeasure$2.invoke(OuterMeasurablePlaceable.kt:94)
        at androidx.compose.ui.node.OuterMeasurablePlaceable$remeasure$2.invoke(OuterMeasurablePlaceable.kt:93)
        at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:1709)
        at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:116)
        at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:75)
        at androidx.compose.ui.node.OwnerSnapshotObserver.observeMeasureSnapshotReads$ui_release(OwnerSnapshotObserver.kt:63)
        at androidx.compose.ui.node.OuterMeasurablePlaceable.remeasure-BRTryo0(OuterMeasurablePlaceable.kt:93)
        at androidx.compose.ui.node.LayoutNode.remeasure-BRTryo0$ui_release(LayoutNode.kt:1201)
        at androidx.compose.ui.node.LayoutNode.remeasure-BRTryo0$ui_release$default(LayoutNode.kt:1199)
        at androidx.compose.ui.node.MeasureAndLayoutDelegate.doRemeasure-0kLqBqw(MeasureAndLayoutDelegate.kt:172)
        at androidx.compose.ui.node.MeasureAndLayoutDelegate.access$doRemeasure-0kLqBqw(MeasureAndLayoutDelegate.kt:39)
        at androidx.compose.ui.node.MeasureAndLayoutDelegate.measureAndLayout(MeasureAndLayoutDelegate.kt:211)
        at androidx.compose.ui.platform.AndroidComposeView.measureAndLayout(AndroidComposeView.android.kt:435)
        at androidx.compose.ui.platform.AndroidComposeView.dispatchDraw(AndroidComposeView.android.kt:568)
        at android.view.View.draw(View.java:22249)
        at android.view.View.updateDisplayListIfDirty(View.java:21122)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4403)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4376)
        at android.view.View.updateDisplayListIfDirty(View.java:21082)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4403)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4376)
        at android.view.View.updateDisplayListIfDirty(View.java:21082)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4403)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4376)
        at android.view.View.updateDisplayListIfDirty(View.java:21082)
        at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4403)
        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4376)
        at android.view.View.updateDisplayListIfDirty(View.java:21082)
        at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:559)
        at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:565)
        at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:642)
        at android.view.ViewRootImpl.draw(ViewRootImpl.java:4031)
        at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3758)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3031)
2021-02-25 11:13:15.315 5798-5798/androidx.compose.samples.crane E/AndroidRuntime:     at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1906)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8089)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:972)
        at android.view.Choreographer.doCallbacks(Choreographer.java:796)
        at android.view.Choreographer.doFrame(Choreographer.java:731)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
        at android.os.Handler.handleCallback(Handler.java:907)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7478)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941)

**Comment by karan4c6** [comment]
The Crane app crashes, while randomly navigating through the top menus

**Comment by yasincidem** [comment]
https://issuetracker.google.com/issues/180488877
This is already reported on Issue Tracker and assigned. 

You can see this bug ignored in HomeTest

```
@Ignore("Swipeable bug makes the app crash: https://issuetracker.google.com/180488877")
    @Test
    fun home_navigatesToAllScreens() {
        composeTestRule.onNodeWithText("Explore Flights by Destination").assertIsDisplayed()
        composeTestRule.onNodeWithText("SLEEP").performClick()
        composeTestRule.onNodeWithText("Explore Properties by Destination").assertIsDisplayed()
        composeTestRule.onNodeWithText("EAT").performClick()
        composeTestRule.onNodeWithText("Explore Restaurants by Destination").assertIsDisplayed()
        composeTestRule.onNodeWithText("FLY").performClick()
        composeTestRule.onNodeWithText("Explore Flights by Destination").assertIsDisplayed()
    }

```

**Comment by yasincidem** [comment]
By the way, this bug occurs when starting the app on landscape mode as well.

**Comment by beneventolorenzo** [comment]
If I recall correctly Crane uses a custom modifier for the swiping gesture. I wanna point out that this bug is also happening in 1.0.0 using a SwipeToDismiss.

**Comment by riggaroo** [comment]
Fixed in #427
